packages/EasyTest/src/Traits/MessengerAssertionsTrait.php

```diff
@@ -113,22 +112,31 @@ static function (object $message) use ($messageClass, $expectedProperties, $prop
             }
         );
 
-        Assert::assertCount($messagesCount ?? 1, $envelopes);
+        Assert::assertCount($messagesCount, $envelopes);
     }
 
-    public static function consumeAsyncMessages(): void
+    /**
+     * @param array<int, array<class-string<\Throwable>, int|string>> $expectedExceptions
+     */
+    public static function consumeAsyncMessages(array $expectedExceptions = []): void
     {
-        self::runMessengerWorker('async', 'default');
+        if (isset(self::$isInsideSafeCall) && self::$isInsideSafeCall) {
```

Comment:
The `isInsideSafeCall` property is from another trait. We shoud avoid mixing traits as much as possible. Is this really required?

Reply:
Yes, it is a useful feature. This will help avoid hiding bugs in tests.

Reply:
What kind of bugs? Just to understand and capture this in the PR.

Reply:
Explanation added in doc comments.